x-pack/plugins/security_solution/public/siem_migrations/rules/components/siem_migrations_table_context.tsx

```diff
@@ -0,0 +1,188 @@
+import { useGetRuleMigrationsQuery } from '../api/hooks/use_get_rule_migrations';
+
+export interface SiemMigrationsTableState {
+  /**
+   * Available rule migrations ids
+   */
+  migrationsIds: string[];
+  /**
+   * Selected rule migration id
+   */
+  selectedMigrationId: string | undefined;
+  /**
+   * Rule migrations available after applying `filterOptions`
+   */
+  ruleMigrations: RuleMigration[];
+  /**
+   * Currently selected table filter
+   */
+  filterOptions: SiemMigrationsTableFilterOptions;
+  /**
+   * All unique tags for all rule translations
+   */
+  tags: string[];
+  /**
+   * If true then there is no cached data and the query is currently fetching.
+   */
+  isLoading: boolean;
+  /**
+   * Rule rows selected in EUI InMemory Table
+   */
+  selectedRuleMigrations: RuleMigration[];
+}
+
+export interface SiemMigrationsTableActions {
+  setFilterOptions: Dispatch<SetStateAction<SiemMigrationsTableFilterOptions>>;
+  selectRuleMigrations: (rules: RuleMigration[]) => void;
+  openRuleTranslationPreview: (ruleId: RuleSignatureId) => void;
+  onMigrationIdChange: (selectedId?: string) => void;
+}
+
+export interface SiemMigrationsContextType {
+  state: SiemMigrationsTableState;
+  actions: SiemMigrationsTableActions;
+}
+
+const SiemMigrationsTableContext = createContext<SiemMigrationsContextType | null>(null);
+
+interface SiemMigrationsTableContextProviderProps {
+  children: React.ReactNode;
+}
+
+const SIEM_MIGRATIONS_INSTALL_FLYOUT_ANCHOR = 'installSiemMigrationsPreview';
+
+export const SiemMigrationsTableContextProvider = ({
+  children,
+}: SiemMigrationsTableContextProviderProps) => {
+  const [selectedRuleMigrations, setSelectedRuleMigrations] = useState<RuleMigration[]>([]);
+
+  const [filterOptions, setFilterOptions] = useState<SiemMigrationsTableFilterOptions>({
+    filter: '',
+    tags: [],
+  });
+
+  const { data: ruleMigrationsStatsAll, isLoading: isLoadingMigrationsStats } =
+    useGetRuleMigrationsStatsAllQuery();
+  const migrationsIds = useMemo(() => {
+    if (isLoadingMigrationsStats || !ruleMigrationsStatsAll?.length) {
+      return [];
+    }
+    return ruleMigrationsStatsAll
+      .filter((migration) => migration.status === 'finished')
+      .map((migration) => migration.migration_id);
+  }, [isLoadingMigrationsStats, ruleMigrationsStatsAll]);
+
+  const [selectedMigrationId, setSelectedMigrationId] = useState<string | undefined>();
+  const onMigrationIdChange = (selectedId?: string) => {
+    setSelectedMigrationId(selectedId);
+  };
+
+  useEffect(() => {
+    if (!migrationsIds.length) {
+      return;
+    }
+    const index = migrationsIds.findIndex((id) => id === selectedMigrationId);
+    if (index === -1) {
+      setSelectedMigrationId(migrationsIds[0]);
+    }
+  }, [migrationsIds, selectedMigrationId]);
+
+  const { data: ruleMigrations, isLoading: isLoadingRuleMigrations } =
+    useGetRuleMigrationsQuery(selectedMigrationId);
```

Comment:
Why don't we call this inside the component that uses this data? And we have a defined `selectedMigrationId`, this can still be _undefined_ here

Reply:
done